hbase-server/src/main/java/org/apache/hadoop/hbase/wal/AbstractRecoveredEditsOutputSink.java

```diff
@@ -233,7 +251,10 @@ void writeRegionEntries(List<WAL.Entry> entries) throws IOException {
         incrementNanoTime(System.nanoTime() - startTime);
       } catch (IOException e) {
         e = e instanceof RemoteException ? ((RemoteException) e).unwrapRemoteException() : e;
-        LOG.error(HBaseMarkers.FATAL, "Got while writing log entry to log", e);
+        final String errorMsg =
+          "Failed to write log entry " + currentWalEntry.toString() + " to log";
```

Comment:
Missing null check here?

Reply:
IOException is thrown by `writer.append(logEntry)` and when we reach there, we are quite certain that `currentWalEntry` will be assigned some value for sure.

Reply:
But this brings a good point, Exception catch should rather be at a place where it is needed, entire block doesn't need to be within try/catch if only single line of code is throwing specific Exception. Let me take care of this.